dissect/cstruct/types/structure.py

```diff
@@ -696,3 +741,83 @@ def __hash__(self):
     """
 
     return dedent(code)
+
+
+def _patch_attributes(func: FunctionType, fields: list[str], start: int = 0) -> FunctionType:
+    """Patches a function's attributes.
+
+    Args:
+        func: The function to patch.
+        fields: List of field names to add.
+        start: The starting index for patching. Defaults to 0.
+    """
+    return type(func)(
+        func.__code__.replace(co_names=(*func.__code__.co_names[:start], *fields)),
+        func.__globals__,
+    )
+
+
+def _generate_structure__init__(fields: list[Field]) -> FunctionType:
+    """Generates an ``__init__`` method for a structure with the specified fields.
+
+    Args:
+        fields: List of field names.
+    """
+    field_names = [field.name for field in fields]
+
+    template: FunctionType = _make_structure__init__(len(field_names))
```

Comment:
To understand this function you need to simultaenously read _make_structure__init__, since we are modifying the code generated there. The naming also suggests that they are strongly related.

Perhaps making `_make_structure__init` an inner function, inside `_generate_structure__init__` is clearer?

Reply:
Ah yes. 

`lru_cache` doesn't support explicitly referencing the cache so that we can make the cache a global variable. 

Ok, nothing to be done.